app/tests/test_token_repository.py

```diff
@@ -0,0 +1,67 @@
+from unittest import TestCase
+from unittest.mock import patch
+
+import boto3
+from docker import from_env, DockerClient
+from docker.models.containers import Container
+
+
+class TestTokenRepository(TestCase):
+    dynamodb_container: Container = None
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.docker_client = from_env()
+        cls.dynamodb_container_name = 'dynamo-test'
+        cls.dynamodb_container = _run_dynamodb_container(cls.docker_client, cls.dynamodb_container_name)
+        try:
+            dynamodb = boto3.resource('dynamodb', endpoint_url='http://localhost:8000')
+            table = dynamodb.create_table(
+                TableName='hallebarde-dev-table',
+                KeySchema=[
+                    {
+                        'AttributeName': 'id',
+                        'KeyType': 'HASH'
+                    }
+                ],
+                AttributeDefinitions=[
+                    {
+                        'AttributeName': 'id',
+                        'AttributeType': 'S'
+                    }
+                ],
+                BillingMode='PAY_PER_REQUEST',
+            )
+
+            table.meta.client.get_waiter('table_exists').wait(TableName='hallebarde-dev-table')
+        except Exception as e:
+            print(e)
+            cls.tearDownClass()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.dynamodb_container.stop()
+        cls.dynamodb_container.remove(v=True)
+
+    @patch('hallebarde.infrastructure.token_repository.boto3.resource')
+    def test_save_should_create_readable_item(self, mock_boto_resource):
```

Comment:
What is an item ? Shouldn't we speak about token instead to be more specific ?

Reply:
I think this triggers a larger question: I do not like the token name, because we indeed create a download token and an upload token for each id. I would rather speak of _exchanges_, what do you think @Mehdi-H ?